packages/@aws-cdk/aws-servicecatalog/test/portfolio.test.ts

```diff
@@ -331,49 +335,55 @@ describe('portfolio associations and product constraints', () => {
   }),
 
-  test('adding identical tag options to portfolio is idempotent', () => {
-    const tagOptions1 = new servicecatalog.TagOptions({
-      key1: ['value1', 'value2'],
-      key2: ['value1'],
-    });
-
-    const tagOptions2 = new servicecatalog.TagOptions({
-      key1: ['value1', 'value2'],
+  test('adding tag options to portfolio multiple times is idempotent', () => {
+    const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+      allowedValuesForTags: {
+        key1: ['value1', 'value2'],
+        key2: ['value1'],
+      },
     });
 
-    portfolio.associateTagOptions(tagOptions1);
-    portfolio.associateTagOptions(tagOptions2); // If not idempotent this would fail
+    portfolio.associateTagOptions(tagOptions);
+    portfolio.associateTagOptions(tagOptions); // If not idempotent this would fail
 
     Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOption', 3); //Generates a resource for each unique key-value pair
     Template.fromStack(stack).resourceCountIs('AWS::ServiceCatalog::TagOptionAssociation', 3);
   }),
 
-  test('fails to add tag options with invalid minimum key length', () => {
-    const tagOptions = new servicecatalog.TagOptions({
-      '': ['value1', 'value2'],
-      'key2': ['value1'],
-    });
+  test('fails to create and then add tag options with invalid minimum key length', () => {
     expect(() => {
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        allowedValuesForTags: {
+          '': ['value1', 'value2'],
+          'key2': ['value1'],
+        },
+      });
+
       portfolio.associateTagOptions(tagOptions);
     }).toThrowError(/Invalid TagOption key for resource/);
   });
 
-  test('fails to add tag options with invalid maxium key length', () => {
-    const tagOptions = new servicecatalog.TagOptions({
-      ['key1'.repeat(1000)]: ['value1', 'value2'],
-      key2: ['value1'],
-    });
+  test('fails to create and then add tag options with invalid maxium key length', () => {
     expect(() => {
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        allowedValuesForTags: {
+          ['key1'.repeat(1000)]: ['value1', 'value2'],
+          key2: ['value1'],
+        },
+      });
+
       portfolio.associateTagOptions(tagOptions);
     }).toThrowError(/Invalid TagOption key for resource/);
   }),
 
-  test('fails to add tag options with invalid value length', () => {
-    const tagOptions = new servicecatalog.TagOptions({
-      key1: ['value1'.repeat(1000), 'value2'],
-      key2: ['value1'],
-    });
+  test('fails to create and then add tag options with invalid value length', () => {
     expect(() => {
+      const tagOptions = new servicecatalog.TagOptions(stack, 'TagOptions', {
+        allowedValuesForTags: {
+          key1: ['value1'.repeat(1000), 'value2'],
+          key2: ['value1'],
+        },
+      });
       portfolio.associateTagOptions(tagOptions);
```

Comment:
I missed the earlier comment about extra lines, but in spirit of trying to keep changes to a minimum, I thought it was fine to keep it now since will be removing them in next PR immediately after this if that's okay.